client/tunnel

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+#
+if [[ $# -eq 0 ]] ; then
+    echo "Please provide the hostname:port as argument, for example $0 host_inside_vpc:3306"
+    exit 0
+fi
+
+
+REGION=eu-central-1
+NAME=bastion-ssm-iam
+DEV_LOCAL_PORT=$(( ( RANDOM % 60000 ) + 1024 ))
+FORWARD_TO=$1
+INSTANCE_ID=$(aws ec2 describe-instances \
```

Comment:
Check if `aws` binary is available:

```bash
if ! [ -x "$(command -v aws)" ]; then
  print_error 'The command aws is required'
  exit 1
fi
```

Reply:
To reduce dependencies one could also use https://github.com/omerh/awsctl

You could download it locally with:

```bash
#!/usr/bin/env bash

set -e
set -u
set -o pipefail

script_directory="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
project_directory="${script_directory}"

temporary_directory="${project_directory}/.tmp"
mkdir -p "${temporary_directory}"

awsctl_path="${temporary_directory}/awsctl"

# Return operating system/platform
function platform {
  os=$(uname)
  case "${os}" in
    'Linux')     echo 'Linux' ;;
    'FreeBSD')   echo 'Freebsd' ;;
    'Darwin')    echo 'Darwin' ;;
    'WindowsNT') echo 'Windows' ;;
    'SunOS')     echo 'Solaris' ;;
    *)           echo 'Unknown' ;;
  esac
}

# Return architecture
function architecture {
  arch=$(uname -m)
  case "${arch}" in
    'i386')   echo 'i386' ;;
    'i686')   echo 'i686' ;;
    'x86_64') echo 'x86_64' ;;
    'arm')    dpkg --print-architecture | grep -q 'arm64' && echo 'arm64' || echo 'arm' ;;
    *)        echo 'unknown' ;;
  esac
}

# Get information about the system
os=$(platform)
arch=$(architecture)


# Download AWS CLI tool: https://github.com/omerh/awsctl
function download_awsctl {
  awsctl_version='0.0.4'

  if [ ! -f "${awsctl_path}" ]; then
    # Download the binary and checksum files.
    curl -OSsL "https://github.com/omerh/awsctl/releases/download/v${awsctl_version}/awsctl_${awsctl_version}_${os}_${arch}.tar.gz"
    curl -OSsL "https://github.com/omerh/awsctl/releases/download/v${awsctl_version}/checksums.txt"

    grep "awsctl_${awsctl_version}_${os}_${arch}.tar.gz" "checksums.txt" > "awsctl_${awsctl_version}_${os}_${arch}_SHA256SUMS"

    # Verify the SHASUM matches the binary.
    if shasum -a 256 -c "awsctl_${awsctl_version}_${os}_${arch}_SHA256SUMS";then
      echo 'awsctl checksum ok'
    else
      echo 'awsctl checksum invalid'
      exit 1
    fi

    tar -xzf "awsctl_${awsctl_version}_${os}_${arch}.tar.gz"
    mv awsctl "${awsctl_path}"

    # Clean up
    rm "awsctl_${awsctl_version}_${os}_${arch}.tar.gz" \
       'checksums.txt' \
       "awsctl_${awsctl_version}_${os}_${arch}_SHA256SUMS" \
       'LICENSE' \
       'README.md'
  else
    echo 'awsctl already downloaded'
    echo "awsctl path: ${awsctl_path}"
  fi
}

download_awsctl
```

It supports:

```bash
$ .tmp/awsctl help
...
Available Commands:
  create      Create AWS Resource
  delete      Delete AWS Resource
  describe    Describe AWS Resource
  get         Get AWS Resource
  help        Help about any command
  set         Set AWS Resource
  version     Print version of awsctl
```

Reply:
> Check if `aws` binary is available:
> 
> ```shell
> if ! [ -x "$(command -v aws)" ]; then
>   print_error 'The command aws is required'
>   exit 1
> fi
> ```

The more common variant is:
```bash
if  ! command -v aws >/dev/null 2>&1; then
    >&2 echo "Error, whatsoever"
    exit 1
fi
```

Reply:
It's a tool for developers, not for noobs, I leave installing dependencies to themselves, but having an error for when it's not there is a good one. :-)